### ISSUE TITLE
Repair suggestions for accessibility checker heading errors

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#12260]
### Is your proposal related to a problem?

Part of [Content quality checkers #11063](https://github.com/wagtail/wagtail/discussions/11063), and [ATAG Guideline B.3.2: Assist authors in repairing accessibility problems](https://www.w3.org/TR/ATAG20/#gl_b32). We should consider implementing repair suggestions in the accessibility checker, per:

> B.3.2.1 Repair Assistance (WCAG): If checking (see Success Criterion B.3.1.1) can detect that a WCAG 2.0 success criterion is not met, then repair suggestion(s) are provided

This is possible in particular for our accessibility rules relating to headings:

> - empty-heading: This rule checks for headings with no text content.
> - heading-order: This rule checks for incorrect heading order.
> - p-as-heading: This rule checks for paragraphs that are styled as headings.

### Describe the solution you'd like

For heading errors specifically, there is room for us to provide either a semi-automated repair, where we present the user with options and upon selecting one the issue is automatically fixed. Or we can provide very actionable suggestions for the user to manually apply:

- For `empty-heading`, we can suggest removing the heading element or adding text.
- For `heading-order`, we can suggest changing the heading element’s level. To either match the preceding heading’s level, or set to "preceding level + 1".
- For `p-as-heading`, we can suggest switching to a semantic heading element. Where the level would again be either "preceding heading" or "preceding heading + 1".

In all three cases there is room for automation when those errors are detected for headings within rich text, particularly when displaying the errors within the page editor.

### Describe alternatives you've considered

It’s for us to decide whether the repair suggestions should be generic, or custom for the scenario based on appropriate heuristics. And whether it’s appropriate to suggest semi-automated repairs based on the complexity.

Another approach could also be to implement a more generic mapping from errors to field (related: #12045), so we can tell users what change to make and where on the page to make it.

### Additional context

See also:

- #12187
- #12091

### Working on this

This requires further design decisions and UX considerations before we proceed with implementation.
